metrics/http.go

```diff
@@ -0,0 +1,110 @@
+package metrics
+
+import (
+	"fmt"
+	"math/big"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/ChainSafe/ChainBridge/core"
+	msg "github.com/ChainSafe/ChainBridge/message"
+	log "github.com/ChainSafe/log15"
+)
+
+type HttpMetricServer struct {
+	port         int
+	core         *core.Core
+	blockheights map[msg.ChainId]BlockHeightInfo
+}
+
+type BlockHeightInfo struct {
+	height      *big.Int
+	lastUpdated time.Time
+}
+
+type httpMetricOptions struct {
+	port int
+	core *core.Core
+}
+
+type chain struct {
+	id int
+}
+
+func newHTTPMetricServer(opts httpMetricOptions) *HttpMetricServer {
+	return &HttpMetricServer{
+		port:         opts.port,
+		core:         opts.core,
+		blockheights: make(map[msg.ChainId]BlockHeightInfo),
+	}
+}
+
+// Start starts the http metrics server
+func (s HttpMetricServer) Start() {
+	log.Info("Metrics server started", "port", s.port)
+
+	// Setup routes
+	http.HandleFunc("/health", s.healthStatus)
+
+	// Start http server
+	// TODO Push strconv to cli parser
+	http.ListenAndServe(":"+strconv.Itoa(s.port), nil)
+}
+
+// healthStatus is a catch-all update that grabs the latest updates on the running chains
+// It assumes that the configuration was set correctly, therefore the relevant chains are
+// only those that are in the core.Core registry.
+func (s HttpMetricServer) healthStatus(w http.ResponseWriter, r *http.Request) {
+
+	// Grab all chains
+	chains := s.core.GetChains()
+	requestTime := time.Now()
+
+	// Iterate through their block heads and update the cache accordingly
+	for _, chain := range chains {
+		latestHeight, err := chain.GetLatestBlock()
+		if err != nil {
+			// NOTE this may error if the chain hasn't been polled yet.
+			// TODO handle re-try
```

Comment:
Probably no need to retry here. Since the blockstore data will not be present only in the first 1 min of the component starting from fresh state. The service calling this health-check will take care of retrying n times.

Reply:
that makes sense!